analyses/mutational-signatures/scripts/de_novo_signature_extraction.R

```diff
@@ -0,0 +1,124 @@
+# Script by J. Taroni for ALSF CCDL
+# Adapted from code written by Anna R. Poetsch and Candace L. Savonen
+# 
+# Given a MAF file, extract de novo mutational signatures for a range of k as
+# specified by the nsignatures_floor and nsignatures_ceiling arguments using 
+# the sigfit package.
+# 
+# This script is essentially a wrapper for sigfit::extract_signatures().
+# 
+# Additional arguments control the number of iterations, the seed, the output
+# RDS file and, optionally, the full path for a PDF of the goodness-of-fit plot
+# from sigfit::plot_gof().
+# 
+# COMMAND LINE USAGE:
+# 
+### To run for a range of values to first identify the number of signatures to
+### use, using a low number of iterations
+#
+# Rscript --vanilla \
+#   scripts/de_novo_signature_extraction.R \
+#   --maf_file ../../data/pbta-snv-consensus-mutation.maf.tsv.gz \
+#   --nsignatures_floor 5 \
+#   --nsignatures_ceiling 15 \
+#   --num_iterations 1000 \
+#   --seed 42 \
+#   --output_file results/denovo_sigfit_signatures.RDS
+#   
+### To run once the number of signatures has been selected, in this example 10,
+### with an appropriate number of iterations
+#
+# Rscript --vanilla \
+#   scripts/de_novo_signature_extraction.R \
+#   --maf_file ../../data/pbta-snv-consensus-mutation.maf.tsv.gz \
+#   --nsignatures_floor 10 \
```

Comment:
In this scenario, it feels a bit odd to supply two numbers for what becomes just one number, but I'm not sure its worth adding an option for.

Reply:
We will need to add a step where we run this for a single value for `nsignatures` once we've picked that value, so that might be a good time to revisit this!